actionlint.cjs

```diff
@@ -8,54 +8,75 @@ require("./wasm_exec.js");
 
 /**
  * @param {WasmLoader} loader
- * @returns {RunActionlint}
+ * @returns {Promise<RunActionlint>}
  */
-module.exports.createActionlint = function createActionlint(loader) {
+module.exports.createActionlint = async function createActionlint(loader) {
+  const encoder = new TextEncoder();
+  const decoder = new TextDecoder();
   const go = new Go();
 
-  /** @type {(() => void)[] | undefined} */
-  let queued = undefined;
+  const wasm = await loader(go);
+  // Do not await this promise, because it only resolves once the go main()
+  // function has exited. But we need the main function to stay alive to be
+  // able to call the `runActionlint` function.
+  go.run(wasm.instance);
 
-  // This function gets called from go once the wasm module is ready and it
-  // executes the linter for all queued calls.
-  globalThis.actionlintInitialized = () => {
-    queued?.forEach((f) => f());
-    queued = globalThis.actionlintInitialized = undefined;
-  };
+  if (!(wasm.instance.exports.memory instanceof WebAssembly.Memory)) {
+    throw new Error("Could not get wasm memory");
+  }
+  const memory = wasm.instance.exports.memory;
+
+  if (!(wasm.instance.exports.WasmAlloc instanceof Function)) {
+    throw new Error("Could not get wasm alloc function");
+  }
+  const wasmAlloc = wasm.instance.exports.WasmAlloc;
+
+  if (!(wasm.instance.exports.WasmFree instanceof Function)) {
+    throw new Error("Could not get wasm free function");
+  }
+  const wasmFree = wasm.instance.exports.WasmFree;
 
-  loader(go).then((wasm) => {
-    // Do not await this promise, because it only resolves once the go main()
-    // function has exited. But we need the main function to stay alive to be
-    // able to call the `runActionlint` function.
-    go.run(wasm.instance);
-  });
+  if (!(wasm.instance.exports.RunActionlint instanceof Function)) {
+    throw new Error("Could not get wasm runActionLint function");
+  }
+  const runActionlint = wasm.instance.exports.RunActionlint;
```

Comment:
I would suggest to destructure objects from `wasm.instance.exports` for readability